language/testing-infra/transactional-test-runner/src/vm_test_harness.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::Path;
+
+use crate::{
+    define_commands,
+    framework::{run_test_impl, CompiledState, MoveAdaptor},
+    tasks::{EmptySubcommand, SyntaxChoice, TaskInput},
+};
+use anyhow::*;
+use either::Either;
+use move_binary_format::{errors::VMResult, file_format::CompiledScript, CompiledModule};
+use move_command_line_common::files::MOVE_IR_EXTENSION;
+use move_core_types::{
+    account_address::AccountAddress,
+    identifier::IdentStr,
+    language_storage::{ModuleId, StructTag, TypeTag},
+    transaction_argument::{convert_txn_args, TransactionArgument},
+};
+use move_lang::{compiled_unit::CompiledUnit, FullyCompiledProgram};
+use move_vm_runtime::{data_cache::MoveStorage, move_vm::MoveVM, session::Session};
+use move_vm_test_utils::InMemoryStorage;
+use move_vm_types::gas_schedule::GasStatus;
+use once_cell::sync::Lazy;
+use resource_viewer::MoveValueAnnotator;
+
+define_commands!(TaskCommand);
+
+const STD_ADDR: AccountAddress =
+    AccountAddress::new([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]);
+
+struct SimpleVMAdaptor<'a> {
+    compiled_state: CompiledState<'a>,
+    storage: InMemoryStorage,
+    default_syntax: Option<SyntaxChoice>,
+}
+
+impl<'a> MoveAdaptor<'a> for SimpleVMAdaptor<'a> {
+    type Subcommand = EmptySubcommand;
+
+    fn compiled_state(&mut self) -> &mut CompiledState<'a> {
+        &mut self.compiled_state
+    }
+
+    fn default_syntax(&self) -> SyntaxChoice {
+        self.default_syntax.unwrap()
+    }
+
+    fn init(pre_compiled_deps: Option<&'a FullyCompiledProgram>) -> Self {
+        Self {
+            compiled_state: CompiledState::new(pre_compiled_deps),
+            default_syntax: None,
```

Comment:
Yeah I think `init` will be necessary for Genesis in the diem side of things, but I could be wrong. regardless not sure bout the `new` comment, basically the same as init with a different name

Reply:
But I'll take a look at cleaning up the default syntax thing

Reply:
> init will be necessary for Genesis in the diem side of things

This didn't seem to be the case to me. `fn run_test_impl` is the entry point of the harness, and it takes an adapter that's already initialized. It would have made sense to me to have `fn init` being part of the trait if it is called by the framework in some way, which it isn't. It's really weird that the trait is telling me how I should implement a function I use internally. I can definitely see adapters which doesn't want the `Option<&'a FullyCompiledProgram>` or wants to have other parameters in the signature of its constructor.

Reply:
> I can definitely see adapters which doesn't want to take an Option<&'a FullyCompiledProgram> or wants to have other parameters in the signature of its constructor.

What adapters are these? we are building a testing framework for just the VM and Diem, not the future of Move

Reply:
> I can definitely see adapters which doesn't want to take an Option<&'a FullyCompiledProgram> or wants to have other parameters in the signature of its constructor.

I guess it's better to say this is just the surface of the problem, or one particular way it is exemplified. The key of my argument here is that, the adapter trait shouldn't really care what my constructor looks like, or whether it exists at all. By the time I hand over control to your framework, I pass you an adapter that's already created and it doesn't matter how I created it. It's not like the framework needs to create new adapters on the fly. Therefore I think we should just delete this function from the trait.